File: CareConnect/CareConnect.Models/Responses/Review.cs
using System;     

namespace CareConnect.Models.Responses
{
    public class Review
    {
        public int ReviewId { get; set; }

        public string Title { get; set; } = null!;

        public string Content { get; set; } = null!;

        public bool IsHidden { get; set; }

        public DateTime PublishDate { get; set; }

        public int? Stars { get; set; }

        public DateTime ModifiedDate { get; set; }

        public virtual Employee? Employee { get; set; }

        public virtual User User { get; set; } = null!;
    }
}    
    
   

File: UI/careconnect_admin/lib/models/responses/review.dart
import 'package:careconnect_admin/models/responses/employee.dart';
import 'package:careconnect_admin/models/responses/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  final int reviewId;
  final String title;
  bool isHidden;
  final String content;
  final DateTime publishDate;
  final int? stars;
  final DateTime modifiedDate;
  final Employee? employee;
  final User user;

  Review({
    required this.reviewId,
    required this.title,
    required this.isHidden,
    required this.content,
    required this.publishDate,
    this.stars,
    required this.modifiedDate,
    this.employee,
    required this.user,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
